source/extensions/filters/http/file_system_buffer/filter_config.cc

```diff
@@ -0,0 +1,202 @@
+    }
+  }
+  throw(EnvoyException("FileSystemBufferFilterConfig must have manager_config"));
+}
+
+const BufferBehavior& getBufferBehavior(
+    const absl::InlinedVector<std::reference_wrapper<const ProtoStreamConfig>, 4>& configs) {
+  for (const ProtoStreamConfig& cfg : configs) {
+    if (cfg.has_behavior()) {
+      return selectBufferBehavior(cfg.behavior());
+    }
+  }
+  return BufferBehaviorStreamWhenPossible::instance;
+}
+
+const FileSystemBufferFilterMergedConfig::StreamConfig getBufferRequestConfig(
+    const absl::InlinedVector<std::reference_wrapper<const FileSystemBufferFilterConfig>, 4>&
+        configs) {
+  absl::InlinedVector<std::reference_wrapper<const ProtoStreamConfig>, 4> request_configs;
+  for (const FileSystemBufferFilterConfig& cfg : configs) {
+    if (cfg.proto().has_request()) {
+      request_configs.emplace_back(cfg.proto().request());
+    }
+  }
+  return FileSystemBufferFilterMergedConfig::StreamConfig(request_configs);
+}
+
+const FileSystemBufferFilterMergedConfig::StreamConfig getBufferResponseConfig(
+    const absl::InlinedVector<std::reference_wrapper<const FileSystemBufferFilterConfig>, 4>&
+        configs) {
+  absl::InlinedVector<std::reference_wrapper<const ProtoStreamConfig>, 4> response_configs;
+  for (const FileSystemBufferFilterConfig& cfg : configs) {
+    if (cfg.proto().has_response()) {
+      response_configs.emplace_back(cfg.proto().response());
+    }
+  }
+  return FileSystemBufferFilterMergedConfig::StreamConfig(response_configs);
+}
+
+} // namespace
+
+FileSystemBufferFilterMergedConfig::StreamConfig::StreamConfig(
+    const absl::InlinedVector<std::reference_wrapper<const ProtoStreamConfig>, 4>& configs)
+    : memory_buffer_bytes_limit_(getMemoryBufferBytesLimit(configs)),
+      storage_buffer_bytes_limit_(getStorageBufferBytesLimit(configs)),
+      storage_buffer_queue_high_watermark_bytes_(getStorageBufferQueueHighWatermarkBytes(configs)),
+      behavior_(getBufferBehavior(configs)) {}
+
+FileSystemBufferFilterConfig::FileSystemBufferFilterConfig(
+    std::shared_ptr<AsyncFileManagerFactory> factory,
+    std::shared_ptr<AsyncFileManager> async_file_manager,
+    const ProtoFileSystemBufferFilterConfig& config)
+    : factory_(std::move(factory)), async_file_manager_(async_file_manager), config_(config) {
+  if (!config_.storage_buffer_path().value().empty()) {
+    struct stat s;
+    if (stat(config_.storage_buffer_path().value().c_str(), &s) != 0 ||
+        (s.st_mode & S_IFDIR) == 0) {
+      throw EnvoyException(
+          absl::StrCat("FileSystemBufferFilterConfig: configured storage_buffer_path '",
+                       config_.storage_buffer_path().value(), "' is not a directory"));
+    }
+  }
+}
+
+FileSystemBufferFilterMergedConfig::FileSystemBufferFilterMergedConfig(
+    const absl::InlinedVector<std::reference_wrapper<const FileSystemBufferFilterConfig>, 4>&
+        configs)
+    : async_file_manager_(getAsyncFileManager(configs)),
```

Comment:
This throw here is not going to work on the data path, you are going to have to actually return a 503, skip caching, etc. so I think you will need to refactor this to actually restart a StatusOr or something in that case so that you can actually handle the error in the filter.

Reply:
Modified so the config just returns a config that has no async_file_manager_, and zeroes out the storage config size so it would, if unchecked, work as a memory-buffer-only implementation. The filter (in the next PR) checks this and returns a 500 if the configuration lacks a manager_config. (Internal server error seems more appropriate than 503 which is supposed to be transient.)